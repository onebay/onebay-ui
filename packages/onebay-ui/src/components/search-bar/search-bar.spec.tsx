import { mount } from '@vue/test-utils'
import SearchBar from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(SearchBar, {
    slots,
    props: { ...values },
  })
}

describe('SearchBar Snap', () => {
  it('render initial SearchBar', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props value', () => {
    const wrapper = factory({ value: 'value' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props placeholder', () => {
    const wrapper = factory({ placeholder: 'placeholder' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props maxLength', () => {
    const wrapper = factory({ maxLength: 120 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props fixed', () => {
    const wrapper = factory({ fixed: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props focus', () => {
    const wrapper = factory({ focus: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props showActionButton', () => {
    const wrapper = factory({ showActionButton: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props actionName', () => {
    const wrapper = factory({ actionName: 'actionName' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render SearchBar -- props inputType number', () => {
    const wrapper = factory({ inputType: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('SearchBar Event', () => {
  it('SearchBar onChange', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange })
    wrapper.find('.ob-search-bar .ob-search-bar__input').trigger('input')
    expect(onChange).toBeCalled()
  })

  it('SearchBar onChange', () => {
    const onFocus = jest.fn()
    const wrapper = factory({ onFocus: onFocus })
    wrapper.find('.ob-search-bar .ob-search-bar__input').trigger('focus')
    expect(onFocus).toBeCalled()
  })

  it('SearchBar onBlur', () => {
    const onBlur = jest.fn()
    const wrapper = factory({ onBlur: onBlur })
    wrapper.find('.ob-search-bar .ob-search-bar__input').trigger('blur')
    expect(onBlur).toBeCalled()
  })

  it('SearchBar onActionClick', async () => {
    const onActionClick = jest.fn()
    const wrapper = factory()
    wrapper.find('.ob-search-bar .ob-search-bar__action').trigger('click')
    await wrapper.setProps({ onActionClick: onActionClick })
    wrapper.find('.ob-search-bar .ob-search-bar__action').trigger('click')
    expect(onActionClick).toBeCalled()
  })

  it('SearchBar onClear', async () => {
    const wrapper = factory({ value: '123' })
    wrapper.find('.ob-search-bar .ob-search-bar__clear').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('change')
    await wrapper.setProps({ onClear: () => { } })
    wrapper.find('.ob-search-bar .ob-search-bar__clear').trigger('click')
  })
})
