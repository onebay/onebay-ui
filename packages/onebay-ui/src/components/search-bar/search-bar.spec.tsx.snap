// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Snap render SearchBar -- props actionName 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    actionName
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props className 1`] = `
<div
  class="ob-search-bar test"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props customStyle 1`] = `
<div
  class="ob-search-bar"
  style="color: red;"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props disabled 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      disabled=""
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props fixed 1`] = `
<div
  class="ob-search-bar ob-search-bar--fixed"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props focus 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 0;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 1; margin-right: 0px;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props inputType number 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="number"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props maxLength 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="120"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props placeholder 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        placeholder
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props showActionButton 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 1; margin-right: 0px;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render SearchBar -- props value 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 0;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: hidden;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: flex;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 1; margin-right: 0px;"
  >
    Search
  </div>
</div>
`;

exports[`SearchBar Snap render initial SearchBar 1`] = `
<div
  class="ob-search-bar"
>
  <div
    class="ob-search-bar__input-cnt"
  >
    <div
      class="ob-search-bar__placeholder-wrap"
      style="flex-grow: 1;"
    >
      <div
        class="ob-icon ob-icon-search"
      />
      <div
        class="ob-search-bar__placeholder"
        style="visibility: visible;"
      >
        Search
      </div>
    </div>
    <input
      class="ob-search-bar__input"
      maxlength="140"
      type="text"
    />
    <div
      class="ob-search-bar__clear"
      style="display: none;"
    >
      <div
        class="ob-icon ob-icon-close-circle"
      />
    </div>
  </div>
  <div
    class="ob-search-bar__action"
    style="opacity: 0;"
  >
    Search
  </div>
</div>
`;
